src/main/java/com/blagodarov/ifmo/web/servlets/CalcServlet.java

```diff
@@ -0,0 +1,130 @@
+package com.blagodarov.ifmo.web.servlets;
+
+import java.io.*;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletException;
+import java.util.regex.Pattern;
+
+@WebServlet(
+    name = "CalcServlet",
+    urlPatterns = {"/calc"}
+)
+public class CalcServlet extends HttpServlet {
+    @Override
+    protected void doGet (HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        PrintWriter writer = response.getWriter();
+        
+        final Map<String, String[]> parameterMap = request.getParameterMap();
+
+        String equation = request.getParameter("equation").replace(" ", "");
+        
+        while (!Pattern.matches("^[0-9*+-/()]+$", equation)){
+            for (Map.Entry<String, String[]> parameterEntry : parameterMap.entrySet()){
+                equation = equation.replace(parameterEntry.getKey(), parameterEntry.getValue()[0]);
+            }
+        }
+        writer.println(calc(equation));
+        writer.flush();
+        writer.close();
+    }
+    private static int calc(String expression) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'calc(String)' has an NPath complexity of 722](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4365389)